noxfile.py

```diff
@@ -0,0 +1,169 @@
+        with environment(env_vars) as env:
+            yield cfg, env
+
+
+@nox.session
+def verify(session):
+    """Prepare and run all available tests"""
+    session.notify(find_session_runner(session, "db-start"))
+    for connector in Settings.CONNECTORS:
+        session.notify(
+            find_session_runner(session, f"integration(connector='{connector}')")
+        )
+    session.notify(find_session_runner(session, "db-stop"))
+
+
+@nox.session(name="db-start", reuse_venv=True)
+def start_db(session):
+    """Start the test database"""
+
+    def start():
+        # Consider adding ITDE as dev dependency once ITDE is packaged properly
+        with session.chdir(Settings.ITDE):
+            session.run(
+                "bash",
+                "start-test-env",
+                "spawn-test-environment",
+                "--environment-name",
+                f"{Settings.ENVIRONMENT_NAME}",
+                "--database-port-forward",
+                f"{Settings.DB_PORT}",
+                "--bucketfs-port-forward",
+                f"{Settings.BUCKETFS_PORT}",
+                "--db-mem-size",
+                "4GB",
+                external=True,
+            )
+
+    def populate():
+        with odbcconfig():
+            settings = {
+                "driver": "EXAODBC",
+                "server": "localhost:8888",
+                "user": "sys",
+                "password": "exasol",
+            }
+            transaction(
+                connect(
+                    "".join(
+                        [
+                            "DRIVER={driver};",
+                            "EXAHOST={server};",
+                            "UID={user};",
+                            "PWD={password}",
+                        ]
+                    ).format(**settings)
+                ),
+                (
+                    "CREATE SCHEMA TEST_SCHEMA;",
+                    "CREATE SCHEMA TEST_SCHEMA_2;",
+                ),
+            )
+
+    start()
+    populate()
+
+
+@nox.session(name="db-stop", reuse_venv=True)
+def stop_db(session):
+    """Stop the test database"""
+    session.run("docker", "kill", "db_container_test", external=True)
+    session.run("docker", "kill", "test_container_test", external=True)
+
+
+@nox.session
+@nox.parametrize("connector", Settings.CONNECTORS)
+def integration(session, connector):
+    """Run(s) the integration tests for a specific connector. Expects a test database to be available."""
+
+    with odbcconfig() as (config, env):
+        connector = (session.posargs or ["pyodbc"])[0]
+        uri = "".join(
+            [
+                "exa+{connector}:",
+                "//sys:exasol@localhost:{db_port}",
+                "/TEST?CONNECTIONLCALL=en_US.UTF-8&DRIVER=EXAODBC",
+            ]
+        ).format(connector=connector, db_port=Settings.DB_PORT)
+        session.run("pytest", "--dropfirst", "--dburi", uri, external=True, env=env)
```

Comment:
I think, I like nox, but I am not sure, how well it cooperates with poetry. Not yet an issue currently for this project, but in general.

Reply:
depends on how and what the use case exactly is, we definitely need to check and align it with poetry.